mars_api/models.py

```diff
@@ -1,8 +1,109 @@
 from django.contrib.auth.models import User
 from django.db import models
+from django.utils import timezone
+from model_utils import Choices
+
+CORPORATIONS = Choices(
+    "Aphrodite",
+    "Aridor",
+    "Arklight",
+    "Celestic",
+    "Cheung Shing Mars",
+    "Credicor",
+    "Ecoline",
+    "Helion",
+    "Interplanetary Cinematics",
+    "Inventrix",
+    "Manutech",
+    "Mining Guild",
+    "Morning Star Inc.",
+    "Phobolog",
+    "Point Luna",
+    "Polyphemos",
+    "Poseidon",
+    "Robinson Industries",
+    "Saturn Systems",
+    "Storm Craft Incorporated",
+    "Terractor",
+    "Tharsis Republic",
+    "Thorgate",
+    "United Nations Mars Initiative",
+    "Valley Trust",
+    "Viron",
+    "Vitor",
+)
+
+
+MAPS = Choices("Tharsis", "Elysium", "Hellas")
 
 
 class Player(models.Model):
     user = models.OneToOneField(User, on_delete=models.CASCADE, null=True, blank=True)
     nickname = models.CharField(max_length=32, unique=True)
     motto = models.CharField(max_length=100, blank=True, null=True)
+
+    def __str__(self):
+        return f"{self.nickname}"
+
+
+class Game(models.Model):
+    date = models.DateTimeField(default=timezone.now)
+    game_map = models.CharField(choices=MAPS, default=MAPS.Tharsis, max_length=14)
+
+    draft_variant = models.BooleanField(default=True)
+    prelude = models.BooleanField(default=False)
+    venus_next = models.BooleanField(default=False)
+    colonies = models.BooleanField(default=False)
+
+    def __str__(self):
+        return f"id: {self.id}; date: {self.date}; game_map: {self.game_map}"
+
+    class Meta:
+        constraints = [
+            models.CheckConstraint(
+                check=models.Q(game_map__in=[map_[0] for map_ in dict(MAPS).items()]),
+                name=f"Only {[map_[0] for map_ in dict(MAPS).items()]} maps are allowed.",
+            )
+        ]
+
+
+class PlayerScore(models.Model):
+    player = models.ForeignKey(Player, models.SET_NULL, related_name="scores", null=True)
```

Comment:
Alternatively you could do `default_related_name = "player_scores"` in the model's `Meta` to make it DRY

Reply:
By non-DRY are you referring to the `related_name` of `game` field? I thought it would be more meaningful to have separate realated names for player and game: `player.scores` reads more concise than `player.player_scores`.

Reply:
They represent exactly the same model on the reverse many-to-one lookup, so I tend to name them the same way for consistency. Also, I feel like having the reverse lookup named _not the same_ as the model would have to be supported by a very strong argument. IMO,
```python
class Score:
    player = models.ForeignKey(Player, models.SET_NULL, null=True)
    game = models.ForeignKey(Game, models.CASCADE)

    class Meta:
         default_related_name = "scores"
```

I'd just go for this to be honest

Reply:
Also, think about using `models.OneToOneField` for player --- score

Reply:
But it's a one-to-many relationship: one player - many scores.

Reply:
:man_facepalming: ignore me :-D